attest/certification.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package attest
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+
+	"github.com/google/go-tpm/tpm2"
+)
+
+// secureCurves represents a set of secure elliptic curves. For now,
+// the selection is based on the key size only.
+var secureCurves = map[tpm2.EllipticCurve]bool{
+	tpm2.CurveNISTP256: true,
+	tpm2.CurveNISTP384: true,
+	tpm2.CurveNISTP521: true,
+	tpm2.CurveBNP256:   true,
+	tpm2.CurveBNP638:   true,
+	tpm2.CurveSM2P256:  true,
+}
+
+// CertificationParameters encapsulates the inputs for certifying an application key.
+// Only TPM 2.0 is supported at this point.
+type CertificationParameters struct {
+	// Public represents the key's canonical encoding (a TPMT_PUBLIC structure).
+	// It includes the public key and signing parameters.
+	Public []byte
+	// CreateData represents the properties of a TPM 2.0 key. It is encoded
+	// as a TPMS_CREATION_DATA structure.
+	CreateData []byte
+	// CreateAttestation represents an assertion as to the details of the key.
+	// It is encoded as a TPMS_ATTEST structure.
+	CreateAttestation []byte
+	// CreateSignature represents a signature of the CreateAttestation structure.
+	// It is encoded as a TPMT_SIGNATURE structure.
+	CreateSignature []byte
+}
+
+// VerifyOpts specifies options for the key certification's verification.
+type VerifyOpts struct {
+	// Public is the public key used to verify key ceritification.
+	Public crypto.PublicKey
+	// Has is the hash function used for signature verification.
```

Comment:
s/Has/Hash/g

Also where does the user get this hash? From the verifying key? Can you please expand the comment?